### ISSUE TITLE
Error

### ISSUE THREAD

**Issue by ghost** [NdT3Development/DiscordBot#22]
Uncaught Exception:  TypeError: guildSettings.swearWords.someword is not a function
how this possible ?

**Comment by TrueWinter** [comment]
Just tested the swear word blocking part of the bot and it seems to be working fine for me. If possible, can you give me some more information such as:
- Where the error occurred,
- What was done before it occurred
- What you have in the `config.js` file for the `swearWords` setting
- What version of Node.js you are running

**Comment by TrueWinter** [comment]
@donkeymonkey1 As there has not been any reply in 10 days from my last reply, and there does not seem to be any problem found with the code while testing, this issue will be closed. If a reply is received, the issue will be reopened.

**Comment by TrueWinter** [comment]
After more testing, without knowing exactly what the issue is from your side, I think I may have found the problem. My testing showed that having a value other than an array for the `swearWords` configuration option will cause the error that you received. Data type checking for configuration options may be added into a future version of the bot but it is currently a low priority.

I did not add this type of check into the bot because at the time, I thought it was not necessary. I overlooked an important part of error checking here. As said earlier in this reply, it will be added into a future version but is not high priority.

For this reason, the issue will be reopened and will be listed as a `P2` priority (where `P0` is the highest and `P3` is the lowest).

The testing can be found online at [https://l.ndt3.ml/2EYdqFj](https://l.ndt3.ml/2EYdqFj). The testing was done on an online service called [repl.it](repl.it).

Thank you for the report.

**Comment by ghost** [comment]
It's problem with dashboard
...

**Comment by ghost** [comment]
When someone change anything in dashboard
Uncaught Exception: TypeError: guildSettings.swearWords.someword is not a function
and bot crash

**Comment by TrueWinter** [comment]
Yeah, it is a problem with the dashboard trying to convert a string to an array but failing to do so. The `swearWords` configuration option then gets sent to the configuration options for that guild as a string and as `.some` expects to be used on an array, it gives an error and crashes.

The code that was added to try to overcome this problem a few months ago was quickly written and not properly tested. I only tested it in a way that I knew it would work and didn't test it with values that could fail to work.

Changing this to a `P1` priority bug.
